modules/host/update.go

```diff
@@ -58,27 +104,182 @@ func (h *Host) ProcessConsensusChange(cc modules.ConsensusChange) {
-		// to and underflowed types.BlockHeight.
-		if block.ID() != types.GenesisBlock.ID() {
-			h.blockHeight++
+		for _, block := range cc.AppliedBlocks {
+			// Look for transactions relevant to open storage obligations.
+			for _, txn := range block.Transactions {
+				// Check for file contracts.
+				if len(txn.FileContracts) > 0 {
+					for i := range txn.FileContracts {
+						fcid := txn.FileContractID(uint64(i))
+						so, err := getStorageObligation(tx, fcid)
+						if err != nil {
+							// The storage folder may not exist, or the disk
+							// may be having trouble. Either way, we ignore the
+							// problem. If the disk is having trouble, the user
+							// will have to perform a rescan.
+							continue
+						}
+						so.OriginConfirmed = true
+						err = putStorageObligation(tx, so)
+						if err != nil {
+							continue
+						}
+					}
+				}
+
+				// Check for file contract revisions.
+				if len(txn.FileContractRevisions) > 0 {
+					for _, fcr := range txn.FileContractRevisions {
+						so, err := getStorageObligation(tx, fcr.ParentID)
+						if err != nil {
+							// The storage folder may not exist, or the disk
+							// may be having trouble. Either way, we ignore the
+							// problem. If the disk is having trouble, the user
+							// will have to perform a rescan.
+							continue
+						}
+						so.RevisionConfirmed = true
+						err = putStorageObligation(tx, so)
+						if err != nil {
+							continue
+						}
+					}
+				}
+
+				// Check for storage proofs.
+				if len(txn.StorageProofs) > 0 {
+					for _, sp := range txn.StorageProofs {
+						so, err := getStorageObligation(tx, sp.ParentID)
+						if err != nil {
+							// The storage folder may not exist, or the disk
+							// may be having trouble. Either way, we ignore the
+							// problem. If the disk is having trouble, the user
+							// will have to perform a rescan.
+							continue
+						}
+						so.ProofConfirmed = true
+						err = putStorageObligation(tx, so)
+						if err != nil {
+							continue
+						}
+					}
+				}
+			}
+
+			// Height is not adjusted when dealing with the genesis block because
+			// the default height is 0 and the genesis block height is 0. If adding
+			// the genesis block, height will already be at height 0 and should not
+			// update.
+			if block.ID() != types.GenesisBlock.ID() {
+				h.blockHeight++
+			}
+
+			// Handle any action items relevant to the current height.
+			bai := tx.Bucket(bucketActionItems)
+			heightBytes := make([]byte, 8)
+			binary.BigEndian.PutUint64(heightBytes, uint64(h.blockHeight))
+			existingItems := bai.Get(heightBytes)
+
+			// From the existing items, pull out a storage obligation.
+			obligationIDs := make([]types.FileContractID, len(existingItems)/crypto.HashSize)
+			for i := 0; i < len(existingItems); i += crypto.HashSize {
+				copy(obligationIDs[i/crypto.HashSize][:], existingItems[i:i+crypto.HashSize])
+			}
```

Comment:
Safer to use the encoding package to store/retrieve these:
```go
// queueActionItem
existingItems = append(existingItems, id) // now a slice of IDs, rather than a byte slice
bai.Put(heightBytes, encoding.Marshal(existingItems)

// ProcessConsensusChange
idBytes := bai.Get(heightBytes)
if idBytes == nil {
    // handle error
}
var obligationIDs []types.FileContractID
err = encoding.Unmarshal(idBytes, &obligationIDs)
```